src/Components/Avatar/Avatar.js

```diff
@@ -0,0 +1,72 @@
+
+import React from 'react';
+import 'antd/dist/antd.css';
+import { Upload, Icon, message } from 'antd';
+
+function getBase64(img, callback) {
+    const reader = new FileReader();
+    reader.addEventListener('load', () => callback(reader.result));
+    reader.readAsDataURL(img);
+}
+
+function beforeUpload(file) {
+    const isJpgOrPng = file.type === 'image/jpeg' || file.type === 'image/png';
+    if (!isJpgOrPng) {
+        message.error('You can only upload JPG/PNG file!');
+    }
+    const isLt2M = file.size / 1024 / 1024 < 2;
+    if (!isLt2M) {
+        message.error('Image must smaller than 2MB!');
+    }
+    return isJpgOrPng && isLt2M;
+}
+
+class Avatar extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            loading: false,
+        };
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    handleChange(info) {
+        if (info.file.status === 'uploading') {
+            this.setState({ loading: true });
+            return;
+        }
+        if (info.file.status === 'done') {
+            // Get this url from response in real world.
+            getBase64(info.file.originFileObj, (imageUrl) => this.setState({
+                imageUrl,
+                loading: false,
+            }));
+        }
+    }
+
+    render() {
+        const { loading } = this.state;
+        const uploadButton = (
+            <div>
+                <Icon type={loading ? 'loading' : 'plus'} />
+                <div className="ant-upload-text">Upload</div>
+            </div>
+        );
+        const { imageUrl } = this.state;
+        return (
+            <Upload
+                name="avatar"
+                listType="picture-card"
+                className="avatar-uploader"
+                showUploadList={false}
+                action="https://www.mocky.io/v2/5cc8019d300000980a055e76"
```

Comment:
URLS and other things like that, could be constants, and place a separate file.

Reply:
Agreed. I'll change it.